src/EditorFeatures/Core/DocumentationComments/DocumentationCommentSuggestion.cs

```diff
@@ -65,26 +65,39 @@ public override Task OnProposalUpdatedAsync(SuggestionSessionBase session, Propo
             return Task.CompletedTask;
         }
 
-        public async Task TryDisplaySuggestionAsync(CancellationToken cancellationToken)
+        public async Task<bool> StartSuggestionSessionAsync(CancellationToken cancellationToken)
         {
-            _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false);
+            await RunWithEnqueueActionAsync(
+                "StartWork",
+                async () => _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false),
+            cancellationToken).ConfigureAwait(false);
 
-            if (_suggestionSession != null)
+            if (_suggestionSession is null)
             {
-                var success = await TryDisplayProposalAsync(_suggestionSession, cancellationToken).ConfigureAwait(false);
-                if (success)
-                {
-                    Logger.Log(FunctionId.Copilot_Generate_Documentation_Displayed, logLevel: LogLevel.Information);
-                }
+                await DisposeAsync().ConfigureAwait(false);
```

Comment:
something smells here in that the disposing seems spread into mulitple different locations.  but i don't have a good sense of what is going on to know where it could best live.

Reply:
talked offline, added comments.